ott/core/bar_problems.py

```diff
@@ -177,3 +191,171 @@ def weights(self) -> jnp.ndarray:
     if self.debiased:
       weights = jnp.concatenate((weights, jnp.array([-0.5])))
     return weights
+
+
+# TODO(michalk8): add citations
+@jax.tree_util.register_pytree_node_class
+class GWBarycenterProblem(BarycenterProblem):
+  """Gromov-Wasserstein barycenter problem, possibly fused.
+
+  Args:
+    y: Array of shape ``[num_measures, N, D]`` containing all points as point
+      clouds. Alternatively, stacked array of shape ``[num_total_points, D]``
+      can also be specified that will be reshaped to ``[num_measures, N, D]``
+      where ``N`` larger or equal to the maximum number of points within all
+      measures. See :class:`~ott.core.bar_problems.BarycenterProblem` or
+      :func:`~ott.core.segment.segment_point_cloud` for more information.
+    b: Array of shape ``[num_measures, N]`` containing the weights
+      (within each measure) of all the points.
+    weights: weights of the barycenter problem (size num_segments).
+    cost: Alternative to ``y``, an array of shape ``[num_measures, N, N]`` that
+      defines padded cost matrices for each measure.
+      Only one of ``y`` and ``cost`` can be passed.
+    y_fused: Array of shape ``[num_measures, N, D_f]`` containing the features
+      of all points used to define the linear term in the fused case.
+      Similarly to ``y``, can be specified as a stacked array of shape
+      ``[num_total_points, D_f]``.
+    loss: Gromov-Wasserstein loss.
+    fused_penalty: Multiplier of the linear term in Fused Gromov-Wasserstein.
+      Only used when ``y_fused != None``.
+    scale_cost: Scaling passed to geometries.
+    kwargs: Keyword arguments for
+      :class:`ott.core.bar_problems.BarycenterProblem`.
+  """
+
+  def __init__(
+      self,
+      y: Optional[jnp.ndarray] = None,
+      b: Optional[jnp.ndarray] = None,
+      weights: Optional[jnp.ndarray] = None,
+      cost: Optional[jnp.ndarray] = None,
```

Comment:
i think it might be more readable that cost comes right after y. maybe use costs. ys is ugly and that's why we stick to y, but cost in singular is a bit confusing.

Reply:
Agreed, changed `cost` -> `costs.`